pkg/scheduler/internal/queue/events.go

```diff
@@ -89,3 +96,85 @@ var (
 	// UnschedulableTimeout is the event when a pod stays in unschedulable for longer than timeout.
 	UnschedulableTimeout = framework.ClusterEvent{Resource: framework.WildCard, ActionType: framework.All, Label: "UnschedulableTimeout"}
 )
+
+// PodSchedulingPropertiesChange interprets the update of a pod and returns corresponding UpdatePodXYZ event(s).
+// Once we have other pod update events, we should update here as well.
+func PodSchedulingPropertiesChange(newPod *v1.Pod, oldPod *v1.Pod) []framework.ClusterEvent {
+	var events []framework.ClusterEvent
+
+	if labelsChanged(newPod.ObjectMeta, oldPod.ObjectMeta) {
+		events = append(events, PodLabelChange)
+	}
+
+	if podResourceRequestChanged(newPod, oldPod) {
+		events = append(events, PodRequestChange)
+	}
+
+	if len(events) == 0 {
+		events = append(events, AssignedPodUpdate)
+	}
+
+	return events
+}
+
+// NodeSchedulingPropertiesChange interprets the update of a node and returns corresponding UpdateNodeXYZ event(s).
+func NodeSchedulingPropertiesChange(newNode *v1.Node, oldNode *v1.Node) []framework.ClusterEvent {
+	var events []framework.ClusterEvent
+
+	if nodeSpecUnschedulableChanged(newNode, oldNode) {
+		events = append(events, NodeSpecUnschedulableChange)
+	}
+	if nodeAllocatableChanged(newNode, oldNode) {
+		events = append(events, NodeAllocatableChange)
+	}
+	if labelsChanged(newNode.ObjectMeta, oldNode.ObjectMeta) {
+		events = append(events, NodeLabelChange)
+	}
+	if nodeTaintsChanged(newNode, oldNode) {
+		events = append(events, NodeTaintChange)
+	}
+	if nodeConditionsChanged(newNode, oldNode) {
+		events = append(events, NodeConditionChange)
+	}
+	if nodeAnnotationsChanged(newNode, oldNode) {
+		events = append(events, NodeAnnotationChange)
+	}
```

Comment:
could we do something like this (obviously with better names)
```go
changedFns := [nodeSpecUnschedulableChanged, nodeAllocatableChanged, ...]
for _, fn := range changedFns {
  event := fn(newNode, oldNode)
  if event != nil {
    events = append(events, event)
  }
}
```

 I think this increases readability, reduces repetition, and reduces risk that changed function returns the wrong event.

Also `PodSchedulingPropertyChanges`, can share this code - some common function - `func(newPod, oldPod, []Fns) []framework.ClusterEvent`

Reply:
Put the refactoring together in #126050.